pkg/repositories/transformers/filters.go

```diff
@@ -0,0 +1,81 @@
+package transformers
+
+import (
+	"context"
+
+	"github.com/lyft/datacatalog/pkg/common"
+
+	"github.com/lyft/datacatalog/pkg/manager/impl/validators"
+	"github.com/lyft/datacatalog/pkg/repositories/gormimpl"
+	"github.com/lyft/datacatalog/pkg/repositories/models"
+	datacatalog "github.com/lyft/datacatalog/protos/gen"
+)
+
+const (
+	partitionKeyFieldName   = "key"
+	partitionValueFieldName = "value"
+	tagNameFieldName        = "tag_name"
+)
+
+var comparisonOperatorMap = map[datacatalog.SinglePropertyFilter_ComparisonOperator]common.ComparisonOperator{
+	datacatalog.SinglePropertyFilter_EQUALS: common.Equal,
+}
+
+func FilterToListInput(ctx context.Context, sourceEntity common.Entity, filterExpression *datacatalog.FilterExpression) (models.ListModelsInput, error) {
+	// ListInput is composed of ModelFilters and ModelJoins. Let's construct those filters and joins.
+	modelFilters := make([]models.ModelValueFilter, 0, len(filterExpression.GetFilters()))
+	joinModelMap := make(map[common.Entity]models.ModelJoinCondition)
+	for _, filter := range filterExpression.GetFilters() {
+		modelPropertyFilters, err := constructModelValueFilter(filter)
+		if err != nil {
+			return models.ListModelsInput{}, err
+		}
+
+		modelFilters = append(modelFilters, modelPropertyFilters...)
+		joiningEntity := modelPropertyFilters[0].GetDBEntity()
```

Comment:
why is this always 0-indexed?

Reply:
I'll add a comment. It is because when we make the call to `constructModelValueFilter`, it can return multiple `ModelPropertyFilter`s. Like in the case of filtering a key/val pair, it creates 2. But both of those `ModelPropertyFilter` are always on the same entity, so that's why we can just use the first one `[0]`.

Reply:
Ok I refactored this to be more readable by putting this in another loop that constructs the joins and doesn't need to do `[0]`. Less efficient, but it doesn't matter because filter expressions are so small.